### PR TITLE
azurerm_bot_channels_registration - support streaming_endpoint_enabled

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes https://github.com/hashicorp/terraform-provider-azurerm/issues/17320

image

### CODE REVIEW

File: internal/services/bot/bot_channels_registration_resource.go [hunk 1]
```diff
@@ -146,6 +146,12 @@ func resourceBotChannelsRegistration() *pluginsdk.Resource {
 				ValidateFunc: validate.BotChannelRegistrationIconUrl,
 			},
 
+			"is_streaming_endpoint_enabled": {
```
Comment (stephybun) [inline internal/services/bot/bot_channels_registration_resource.go:149]: Might be better as just `streaming_endpoint_enabled`, WDYT?

**Review by stephybun — CHANGES_REQUESTED** [review]

File: internal/services/bot/bot_channels_registration_resource.go (hunk 1 above)
Comment (neil-yechenwei) [reply internal/services/bot/bot_channels_registration_resource.go]: Sure. Updated

**Comment by neil-yechenwei** [comment]
@stephybun ,thanks for your comment. I've updated code. Please take another look. Thanks in advance.

File: internal/services/bot/bot_channel_test.go [hunk 2]
```diff
@@ -11,9 +11,10 @@ func TestAccBotChannelsRegistration(t *testing.T) {
 	// Azure only being able provision against one app id at a time
 	acceptance.RunTestsInSequence(t, map[string]map[string]func(t *testing.T){
 		"basic": {
-			"basic":    testAccBotChannelsRegistration_basic,
-			"update":   testAccBotChannelsRegistration_update,
-			"complete": testAccBotChannelsRegistration_complete,
+			"basic":                      testAccBotChannelsRegistration_basic,
+			"update":                     testAccBotChannelsRegistration_update,
+			"complete":                   testAccBotChannelsRegistration_complete,
+			"isStreamingEndpointEnabled": testAccBotChannelsRegistration_isStreamingEndpointEnabled,
```
Comment (stephybun) [inline internal/services/bot/bot_channel_test.go:17]: Can we also rename the test to just `streamingEndpointEnabled`?

Comment (neil-yechenwei) [reply internal/services/bot/bot_channel_test.go:17]: Updated.

**Comment by FarihahHamizan** [comment]
Is it possible to do the same changes for azurerm_bot_service_azure_bot resource? @katbyte

**Pull request merged**